app/src/main/java/org/p2p/wallet/jupiter/repository/tokens/JupiterSwapTokensRemoteRepository.kt

```diff
@@ -75,32 +73,48 @@ internal class JupiterSwapTokensRemoteRepository(
         return (now - fetchTokensDate) <= TimeUnit.DAYS.toMillis(1) // check day has passed
     }
 
-    override suspend fun getTokensRates(tokens: List<JupiterSwapToken>): Map<Base58String, TokenPrice> {
+    override suspend fun getTokensRates(tokens: List<JupiterSwapToken>): Map<Base58String, TokenServicePrice> {
         val tokenMints = tokens.map(JupiterSwapToken::tokenMint)
-        val tokensCoingeckoIds = tokens.mapNotNull { it.coingeckoId?.let(::TokenCoinGeckoId) }
 
-        if (tokensCoingeckoIds.isEmpty()) {
+        if (tokenMints.isEmpty()) {
             return emptyMap()
         }
-        val isTokenPricesCachedForMints = tokenMints.all { it in pricesCache }
+        val isTokenPricesCachedForMints =
+            tokenMints.all {
+                tokenServiceInteractor.findTokenPriceByAddress(
+                    tokenAddress = it.base58Value
+                ) != null
+            }
         if (isTokenPricesCachedForMints) {
-            return tokenMints.associateWith(pricesCache::requirePriceByMint)
+            return tokenMints.associateWith {
+                tokenServiceInteractor.fetchTokenPriceByAddress(
```

Comment:
what's the difference between fetch and find token price? seems to me they are the same. 
and also - fetch is more like network related verb, it's better to have find here too